.github/workflows/codemod_publish.yml

```diff
@@ -1,107 +1,82 @@
-name: Codemod publish
+# yaml-language-server: $schema=https://json.schemastore.org/github-workflow.json
+
+# For more information see: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow
+
+name: Publish Codemod
 
 on:
-  workflow_dispatch:
   push:
-    paths:
-      - "recipes/**"
-    branches:
-      - main
-
-permissions:
-  contents: read
+    tags:
+      - 'v*@*'  # Match tags like v1.0.0@my-codemod
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: 'Tag to publish (e.g., v1.0.0@my-codemod)'
+        required: true
+        type: string
 
 jobs:
-  paths-filter:
-    permissions:
-      contents: read
-      pull-requests: read
-    name: Check for .codemodrc.json files changes
+  publish:
     runs-on: ubuntu-latest
-    outputs:
-      codemods: ${{ steps.filter.outputs.codemods }}
-      codemods_files: ${{ steps.filter.outputs.codemods_files }}
+
     steps:
-      - name: Harden the runner (Audit all outbound calls)
-        uses: step-security/harden-runner@6c439dc8bdf85cadbbce9ed30d1c7b959517bc49
-        with:
-          egress-policy: audit
+      - name: Checkout code
+        uses: actions/checkout@v4
```

Comment:
For an abundance of caution, let's use the release hash (dependabot will handle updating it after this is committed with a hash).

Reply:
I think we can do both: first fix the things we know about, and then check for things we don't know about. Otherwise, there is a however brief window of vulnerability to things we already could have fixed.

Reply:
oh that found there ... I'll go to bed a little wiser.